scripts/prepack.ts

```diff
@@ -69,28 +99,54 @@ if (tmpCopyBundles) {
 }
 // end remove
 
-// package.json modifications
-const packageJsonPath = path.resolve(buildDir, 'package.json');
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const pkgJson: { [key: string]: unknown } = require(packageJsonPath);
+const packageJsonPath = path.resolve('package.json');
+const pkgJson: { [key: string]: unknown } = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf8' }));
+
+function getModifiedPkgJson(newName: string, removeDependencies?: boolean): Record<string, unknown> {
+  const newPkgJson = { ...pkgJson };
+  newPkgJson.name = newName;
 
-// modify entry points to point to correct paths (i.e. strip out the build directory)
-ENTRY_POINTS.filter(entryPoint => pkgJson[entryPoint]).forEach(entryPoint => {
-  pkgJson[entryPoint] = (pkgJson[entryPoint] as string).replace(`${buildDir}/`, '');
-});
+  delete newPkgJson.scripts;
+  delete newPkgJson.volta;
+  delete newPkgJson.jest;
 
-delete pkgJson.scripts;
-delete pkgJson.volta;
-delete pkgJson.jest;
+  if (removeDependencies) {
+    newPkgJson.dependencies = {};
+  }
+
+  // modify entry points to point to correct paths (i.e. strip out the build directory)
+  ENTRY_POINTS.forEach(entryPoint => {
+    const oldEntryPoint = pkgJson[entryPoint] as string | undefined;
+    if (oldEntryPoint) {
+      newPkgJson[entryPoint] = path.relative(buildDir, oldEntryPoint);
+    }
+  });
 
-// write modified package.json to file (pretty-printed with 2 spaces)
+  return newPkgJson;
+}
+
+// write modified package.json to new location
 try {
-  fs.writeFileSync(packageJsonPath, JSON.stringify(pkgJson, null, 2));
+  const modifiedPkgJson = getModifiedPkgJson(pkgJson.name as string);
+  const modifiedPkgJsonPath = path.resolve(buildDir, 'package.json');
+  fs.writeFileSync(modifiedPkgJsonPath, JSON.stringify(modifiedPkgJson, null, 2));
 } catch (error) {
   console.error('Error while writing package.json to disk');
   process.exit(1);
 }
 
+// write modified package.json for legacy package to new location
+if (shouldPrepareLegacyPackage) {
+  try {
+    const modifiedPkgJson = getModifiedPkgJson(`${pkgJson.name}-legacy`, true);
+    const modifiedPkgJsonPath = path.resolve(NPM_LEGACY_BUILD_DIR, 'package.json');
+    fs.writeFileSync(modifiedPkgJsonPath, JSON.stringify(modifiedPkgJson, null, 2));
```

Comment:
Totally optional nit: Should we re-add the comment here that explained the `2` in the parameter?

Reply:
No need to explain JS api